components/datetime/src/fields/macros.rs

```diff
@@ -83,13 +94,16 @@ macro_rules! field_type {
             /// This is mostly useful for serialization,
             /// and does not guarantee index stability between ICU4X
             /// versions.
-            pub(crate) fn _from_idx(idx: u8) -> Result<Self, SymbolError> {
-                match idx {
-                    $(
-                        $idx => Ok(Self::$val),
-                    )*
-                    _ => Err(SymbolError::InvalidIndex(idx)),
-                }
+            #[inline]
+            pub(crate) fn from_idx(idx: u8) -> Result<Self, SymbolError> {
+                Self::try_from(idx)
+                    .map_err(|_| SymbolError::InvalidIndex(idx))
```

Comment:
grumble: don't like that these impl TryFrom and not their own trait, but I know you don't like this either 😄